tests/waku_lightpush/test_client.nim

```diff
@@ -0,0 +1,323 @@
+  testutils/unittests,
+  chronicles,
+  chronos,
+  libp2p/crypto/crypto
+
+
+import
+  ../../../waku/[
+    node/peer_manager,
+    waku_core,
+    waku_lightpush,
+    waku_lightpush/common,
+    waku_lightpush/client,
+    waku_lightpush/protocol_metrics,
+    waku_lightpush/rpc
+  ],
+  ../testlib/[
+    wakucore,
+    testasync,
+    futures,
+    testutils
+  ],
+  ./lightpush_utils,
+  ../resources/[
+    pubsub_topics,
+    content_topics,
+    payloads
+  ]
+
+
+suite "Waku Lightpush Client":
+  var
+    handlerFuture {.threadvar.}: Future[(PubsubTopic, WakuMessage)]
+    handler {.threadvar.}: PushMessageHandler
+
+    serverSwitch {.threadvar.}: Switch
+    clientSwitch {.threadvar.}: Switch
+    server {.threadvar.}: WakuLightPush
+    client {.threadvar.}: WakuLightPushClient
+
+    serverRemotePeerInfo {.threadvar.}: RemotePeerInfo
+    clientPeerId {.threadvar.}: PeerId
+    pubsubTopic {.threadvar.}: PubsubTopic
+    contentTopic {.threadvar.}: ContentTopic
+    message {.threadvar.}: WakuMessage
+    
+  asyncSetup:
+    handlerFuture = newPushHandlerFuture()
+    handler = proc(peer: PeerId, pubsubTopic: PubsubTopic, message: WakuMessage): Future[WakuLightPushResult[void]] {.async.} =
+      handlerFuture.complete((pubsubTopic, message))
+      return ok()
+
+    serverSwitch = newTestSwitch()
+    clientSwitch = newTestSwitch()
+    server = await newTestWakuLightpushNode(serverSwitch, handler)
+    client = newTestWakuLightpushClient(clientSwitch)
+
+    await allFutures(serverSwitch.start(), clientSwitch.start())
+
+    serverRemotePeerInfo = serverSwitch.peerInfo.toRemotePeerInfo()
+    clientPeerId = clientSwitch.peerInfo.peerId
+    pubsubTopic = DefaultPubsubTopic
+    contentTopic = DefaultContentTopic
+    message = fakeWakuMessage()
+
+  asyncTeardown:
+    await allFutures(clientSwitch.stop(), serverSwitch.stop())
+
+  suite "Verification of PushRequest Payload":
+    asyncTest "Valid Payload Types":
+      # Given the following payloads
+      let
+        message2 = fakeWakuMessage(payloads.ALPHABETIC, content_topics.CURRENT)
+        message3 = fakeWakuMessage(payloads.ALPHANUMERIC, content_topics.TESTNET)
+        message4 = fakeWakuMessage(payloads.ALPHANUMERIC_SPECIAL, content_topics.PLAIN)
+        message5 = fakeWakuMessage(payloads.EMOJI, content_topics.CURRENT)
+        message6 = fakeWakuMessage(payloads.CODE, content_topics.TESTNET)
+        message7 = fakeWakuMessage(payloads.QUERY, content_topics.PLAIN)
+        message8 = fakeWakuMessage(payloads.TEXT_SMALL, content_topics.CURRENT)
+        message9 = fakeWakuMessage(payloads.TEXT_LARGE, content_topics.TESTNET)
+
+      # When publishing a valid payload
+      let publishResponse = await client.publish(pubsubTopic, message, serverRemotePeerInfo)
+
+      # Then the message is received by the server
+      discard await handlerFuture.withTimeout(FUTURE_TIMEOUT)
+      check:
+        publishResponse.isOk()
+        handlerFuture.finished()
```

Comment:
Nitpick: I think assert is more convenient when dealing with `Result`. The recommendation applies elsewhere.
```suggestion
      assert publishResponse.isOk(), publishResponse.error
      check handlerFuture.finished()
```